### PR TITLE
PHP SDK integration

### LINKED ISSUE #103: [Feature Request] PHP harness
So we can start creating PHP specs

### PR DESCRIPTION
**Pull request by roxblnfk**
## What was changed

Added runner preparation:
- [x] configuring SDK version
- [x] downloading PHP vendor packages
- [x] downloading RoadRunner
- [x] do it in a separated directory

Added runner that runs RoadRunner and starts PHP features in a single run.

Implemented features: 
- [x] activity `3 / 5`
- [ ] build_id_versioning `0 / 6` **_(not supported)_**
  - https://github.com/temporalio/sdk-php/issues/311
- [x] child_workflow `3 / 13`
  - https://github.com/temporalio/sdk-php/issues/451
- [x] client `0 / 2`
- [x] continue_as_new `1 / 1`
  - https://github.com/temporalio/sdk-php/issues/453
- [x] data_converter `7 / 7`
  - https://github.com/temporalio/sdk-php/issues/455
  - https://github.com/temporalio/sdk-php/issues/454
- [ ] eager_activity `0 / 1` _(waiting SDK feature)_
  - https://github.com/temporalio/sdk-php/issues/456
- [x] eager_workflow `1 / 1`
- [x] local_activity `0 / 15` _(all cases are empty)_
- [x] query `5 / 5`
- [x] reset `0 / 1` _(not required)_
- [x] schedule `4 / 5` _(our cron doesn't support seconds. **Should it support seconds?**)_
- [x] search_attributes `0 / 2` _(all cases are empty)_
- [x] signal `6 / 7`
  - https://github.com/temporalio/sdk-php/issues/457
- [x] telemetry `0 / 1` _**(OTEL package is not released yet)**_
- [x] timer `0 / 0`
- [x] tracing `0 / 0`
- [x] update `11 / 11`
  - https://github.com/temporalio/sdk-php/issues/458
  - https://github.com/temporalio/sdk-php/issues/463

## Checklist

- Closes #103
- [ ] CI
  - [ ] Register docker image there https://github.com/temporalio/docker-builds/blob/main/.github/workflows/features-integration.yml
    - [x] https://github.com/temporalio/features/pull/541
    - [x] https://github.com/temporalio/features/pull/546
    - [ ] Wait for merging this PR

## How to run

```bash
go build -o temporal-features
./temporal-features run --lang php
```

Windows:
```bash
go build -o temporal-features.exe
./temporal-features.exe run --lang php
```

### CODE REVIEW

**Comment by roxblnfk** [comment]
Looks like we need to register the `temporaliotest/features:php` docker image

**Comment by roxblnfk** [comment]
About this option:

> - [ ] eager_activity 0 / 1 (waiting SDK feature)
> https://github.com/temporalio/sdk-php/issues/456

Adding this test requires adjustments to the framework because RR needs to be restarted with special worker settings.
I'm not sure it's worth spending time on it right now.

**Comment by roxblnfk** [comment]
Ready for review.
You don't need to spend effort reviewing PHP files.
For the most part, they are the same tests that are also [run in the PHP-SDK](https://github.com/temporalio/sdk-php/tree/master/tests/Acceptance/Harness).

File: sdkbuild/php.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+// BuildPhpProgramOptions are options for BuildPhpProgram.
+type BuildPhpProgramOptions struct {
+	// Required version. If it contains a slash it is assumed to be a path with
+	// a single wheel in the dist directory. Otherwise it is a specific version
```
Comment (Sushisource) [inline sdkbuild/php.go:15]: Copy-paste leftover from Python about wheels

**Review by Sushisource — APPROVED** [review]
Awesome. Having this is super valuable.

File: sdkbuild/php.go (hunk 1 above)
Comment (roxblnfk) [reply sdkbuild/php.go]: Fixed

**Comment by roxblnfk** [comment]
Looks like `system.enableEagerWorkflowStart` is not `true` now.
That's why the `EagerWorkflow` test is failing

**Comment by roxblnfk** [comment]
BTW is it okay that I'm getting TRUE in the `eagerWorkflowStart` field of a `ServerCapabilities` response while the server is started with the option `system.enableEagerWorkflowStart=false`?

**Comment by Sushisource** [comment]
> Looks like `system.enableEagerWorkflowStart` is not `true` now. That's why the `EagerWorkflow` test is failing.
> 
> Might be related with #543

Ah, interesting... yeah could be.

>BTW is it okay that I'm getting TRUE in the eagerWorkflowStart field of a ServerCapabilities response while the server is started with the option system.enableEagerWorkflowStart=false?

Hmm... I think it might be OK even if it's kinda weird. I think the semantics of the API are merely that it *can* support it at that version, not that it's currently enabled... which does seem a little silly.

**Comment by roxblnfk** [comment]
@Sushisource now it works 👍 

It may be necessary to add dynamicconfig to the Dockerfiles of other SDKs in the future, as [done in the PHP Dockerfile](https://github.com/temporalio/features/pull/485/commits/51b0ba344d3d977634e54cbbf470d66acac11fb2). At least for now, the tests there are not failing. 🙂

**Pull request merged**